mission-control/docs/config-db/tutorials/file-scraper.md

```diff
@@ -0,0 +1,154 @@
+# Scraping Custom File
+
+In this tutorial we'll scrape a custom JSON file and see how config-db monitors changes to it over time.
+
+Let's start off by creating this example file that we want to scrape
+
+```json title="vms.json"
+{
+  "servers": [
+    {
+      "name": "web-server",
+      "ip": "192.168.1.100",
+      "port": 80,
+      "role": "frontend"
+    },
+    {
+      "name": "api-server",
+      "ip": "192.168.1.101",
+      "port": 8080,
+      "role": "backend"
+    },
+    {
+      "name": "database-server",
+      "ip": "192.168.1.102",
+      "port": 27017,
+      "role": "database"
+    }
+  ],
+  "logging": {
+    "level": "info",
+    "file_path": "/var/log/app.log"
+  }
+}
+```
+
+Place this file anywhere in your system and take note of its path. For the sake of simplicity, let's keep that file inside the `/tmp` dir.
+
+## Goal
+
+In the file above, we're mainly concerned with the servers. We want to have each of those servers as a config items.
+
+## Designing the scaper
+
+Each config item needs two mandatory fields
+
+- id
+- type
+
+We need to tell config-db where it can get those for each of our servers.
+
+We can utilize the IP address as the identifier since it is unique for our purposes, and designate the "role" attribute as the type.
+We do this by specifying the fields using JSONPath.
+Additionally, we're also going to specify the name of each config even though it's optional.
+
+We can also supply hard coded values instead of a json path. Since all of these servers are virtual machines, let's specify the class as a static value "VritualMachine".
+
+:::note
+If the class wasn't specified, the type would have been used as a fallback.
+:::
+
+Now finally, we need to tell config-db how it can generate 3 different configs for each of those servers from one config file. We do that by using transformation scripts. In this example, we've used javascript but other scripting option are also available like - CEL Expression & Go Template.
+
+```yaml title='vms-scraper.yaml'
+apiVersion: configs.flanksource.com/v1
+kind: ScrapeConfig
+metadata:
+  name: vm-scraper
+  namespace: default
+spec:
+  schedule: '@every 30s'
+  file:
+    - type: $.role
+      id: $.ip
+      name: $.name
+      class: VirtualMachine
+      transform:
```

Comment:
Don't include the transform unless its doing something useful

Reply:
How would you get the 3 config items without using transform in this case?